src/idl_parser.cpp

```diff
@@ -2004,18 +2048,39 @@ CheckedError Parser::DoParse(const char *source, const char **include_paths,
   while (token_ != kTokenEof) {
     if (opts.proto_mode) {
       ECHECK(ParseProtoDecl());
-    } else if (token_ == kTokenNameSpace) {
-      ECHECK(ParseNamespace());
-    } else if (token_ == '{') {
+  }
+  else if (token_ == kTokenNameSpace) {
+    ECHECK(ParseNamespace());
+
+  } else if (token_ == '{' || token_ == '[') {
       if (!root_struct_def_)
         return Error("no root type set to parse json with");
       if (builder_.GetSize()) {
         return Error("cannot have more than one json object in a file");
       }
-      uoffset_t toff;
+
+    uoffset_t toff;
+
+    const bool top_level_list = token_ == '[';
+    if (top_level_list)
+    {
+      if(root_struct_def_->fields.vec.size() > 1)
+        return Error("top level list found with more than one field in schema");
+      
+      auto field = root_struct_def_->fields.vec.front();
+      Value val = field->value;
+      ECHECK(ParseAnyValue(val, field, 0, root_struct_def_));
+      //ECHECK(ProcessTableFields(1, *val.type.struct_def, nullptr, &toff));
+      field_stack_.insert(field_stack_.rbegin().base(), std::make_pair(val, field));
+      ECHECK(ProcessTableFields(1, *root_struct_def_, nullptr, &toff));
+    }
+    else
+    {
       ECHECK(ParseTable(*root_struct_def_, nullptr, &toff));
-      builder_.Finish(Offset<Table>(toff),
-                file_identifier_.length() ? file_identifier_.c_str() : nullptr);
+    }
+    builder_.Finish(Offset<Table>(toff),
```

Comment:
That statement was actually factored out as a consequence of introducing an if/else statement just above and so is part of the changes